packages/graphql-toolbox/src/utils/tracking.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LOCAL_STATE_ENABLE_PRODUCT_USAGE_TRACKING } from "src/constants";
+import { Storage } from "./storage";
+
+class Tracking {
+    public trackFavorites = () => {
+        this.fireTrackingEvent("ux", "favorites");
+    };
+
+    private fireTrackingEvent = (eventCategory: string, eventLabel: string, eventProperties = {}) => {
+        const trackingConsent = Storage.retrieve(LOCAL_STATE_ENABLE_PRODUCT_USAGE_TRACKING);
+
+        if (trackingConsent !== "true") return;
+        if (!window.analytics || !window.analytics.track) return;
+
+        const enrichedEventProperties = {
+            graphQLToolboxVersion: process.env.VERSION,
+            neo4jGraphQLLibraryVersion: process.env.NEO4J_GRAPHQL_VERSION,
+            ...eventProperties,
+        };
```

Comment:
```suggestion
        const enrichedEventProperties = {
            ...eventProperties,
            graphQLToolboxVersion: process.env.VERSION,
            neo4jGraphQLLibraryVersion: process.env.NEO4J_GRAPHQL_VERSION, 
        };
```

Just a thought about the order of the spreading here: I prefer spreading first so defaults get overwritten by anything more specifically added after. It may not functionally change anything here at all but it feels like it could be a nice habit to get into.

Reply:
Good point! I moved the spreading to the bottom to prevent it from being overwritten but I suppose it would be ok if we actually could overwrite the default. changing this